gGRPC error message using JDK 24 GraalVM

### Describe the bug

I used the quarkus project generator from the website and chose `grpc` as one of the extensions.
I then compiled the project and things work fine but I see this error message and not sure if it will cause any problems?

I was trying to use GraalVM 24.0.2 to compile... 
JDK 21 works fine and no message is displayed.

```
2025-08-26 10:23:28,126 INFO  [io.qua.grp.run.sup.Channels] (main) Shutting down gRPC channel ManagedChannelOrphanWrapper{delegate=ManagedChannelImpl{logId=4, target=dns:///0.0.0.0:9001}}
Exception in thread "vert.x-internal-blocking-1" java.lang.IllegalAccessError: module java.base does not open java.lang to unnamed module @679dd234; to use the thread-local-reset capability on Java 24 or later, use this JVM option: --add-opens java.base/java.lang=ALL-UNNAMED
	at org.jboss.threads.JDKSpecific$ThreadAccess.<clinit>(JDKSpecific.java:32)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:13)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1447)
Exception in thread "executor-thread-1" java.lang.NoClassDefFoundError: Could not initialize class org.jboss.threads.JDKSpecific$ThreadAccess
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:13)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1447)
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.IllegalAccessError: module java.base does not open java.lang to unnamed module @679dd234; to use the thread-local-reset capability on Java 24 or later, use this JVM option: --add-opens java.base/java.lang=ALL-UNNAMED [in thread "vert.x-internal-blocking-1"]
	at org.jboss.threads.JDKSpecific$ThreadAccess.<clinit>(JDKSpecific.java:32)
	... 3 more```


### Expected behavior

No message to display

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

 java -version openjdk version "24.0.2" 2025-07-15 OpenJDK Runtime Environment GraalVM CE 24.0.2+11.1 (build 24.0.2+11-jvmci-b01) OpenJDK 64-Bit Server VM GraalVM CE 24.0.2+11.1 (build 24.0.2+11-jvmci-b01, mixed mode, sharing)

### Quarkus version or git rev

3.25.4

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

mvn 3.9.10

### Additional information

_No response_

--- a/extensions/grpc/codegen/src/main/java/io/quarkus/grpc/codegen/GrpcCodeGen.java
+++ b/extensions/grpc/codegen/src/main/java/io/quarkus/grpc/codegen/GrpcCodeGen.java
@@ -48,6 +48,7 @@ import io.smallrye.common.process.ProcessBuilder;
  */
 public class GrpcCodeGen implements CodeGenProvider {
     private static final Logger log = Logger.getLogger(GrpcCodeGen.class);
+    private static final Logger grpcProcessOuputLogger = Logger.getLogger("protoc");
 
     private static final String quarkusProtocPluginMain = "io.quarkus.grpc.protoc.plugin.MutinyGrpcGenerator";
     private static final String EXE = "exe";
@@ -194,7 +195,14 @@ public class GrpcCodeGen implements CodeGenProvider {
                 }
                 log.debugf("Executing command: %s", String.join(" ", command));
                 try {
-                    ProcessBuilder.exec(command.get(0), command.subList(1, command.size()));
+                    ProcessBuilder<Void> pb = ProcessBuilder.newBuilder(command.get(0),
+                            command.subList(1, command.size()));
+                    // Tune the environment for compatibility with Java >24 without triggering warnings
+                    pb.modifyEnvironment(GrpcCodeGen::invocationEnvironmentTuning);
+                    // Set up a custom output handler to highlight only relevant errors
+                    pb.output().consumeLinesWith(100, this::outputConsumer);
+                    pb.error().consumeLinesWith(100, this::outputConsumer).logOnSuccess(false);
+                    pb.run();
                 } catch (Exception e) {
                     throw new CodeGenException("Failed to generate Java classes from proto files: %s to %s with command %s"
                             .formatted(protoFiles, outDir.toAbsolutePath(), String.join(" ", command)), e);
@@ -210,6 +218,37 @@ public class GrpcCodeGen implements CodeGenProvider {
         }
 
         return false;
+    }
+
+    private static void invocationEnvironmentTuning(final Map<String, String> environment) {
+        //This specific environment variable is being picked up by the JVMs spawned by protoc:
+        final String key = "JDK_JAVA_OPTIONS";
+        String existingValue = environment.get(key);
+        if (existingValue == null || existingValue.isBlank()) {
+            existingValue = "";
+        }
+        StringBuilder sb = new StringBuilder();
+        //Each of these require custom logic to ensure we don't override an explicit user setting
+
+        if (!existingValue.contains("-Dsun.stdout.encoding=")) {
+            //This one is always useful, especially on Java 17:
+            sb.append("-Dsun.stdout.encoding=UTF-8 ");
+        }
+
+        //Do NOT set this property on Java 17, as it will fail with "unrecognized option":
+        if (Runtime.version().feature() > 21 && !existingValue.contains("--sun-misc-unsafe-memory-access=")) {
+            sb.append("--sun-misc-unsafe-memory-access=allow ");
+        }
+        sb.append(existingValue);
+        environment.put(key, sb.toString().trim());
+    }
+
+    private void outputConsumer(final String line) {
+        //Protoc will by default emit a message like "NOTE: Picked up JDK_JAVA_OPTIONS: [...],
+        //which will lead the ProcessBuilder to emit a warning if not consumed.
+        //So let's consume them here and emit as regular "info": if there's errors, they will be caught
+        //as the return code is being checked as well.
+        grpcProcessOuputLogger.info(line);
     }
 
     private static void copySanitizedProtoFile(ResolvedDependency artifact, Path protoPath, Path outProtoPath)

